### ISSUE TITLE
📎 Implement `noUndeclaredClasses`

### ISSUE THREAD

**Issue by ematipico** [biomejs/biome#9158]
### Description

A lint rule that detects styles defined in HTML-ish languages, but that aren't declared in the file. In the following example, `bar` will trigger an error

```vue
<template>
  <span class="foo bar"></span>
</template>
<style>
.foo {}
</style>
```

**Comment by dyc3** [comment]
Should this be named `noUndeclaredClasses` or `noUndeclaredCssClasses`?

Also, your rule description seems off. It's describing `noUnusedStyles`

**Comment by ematipico** [comment]
Updated the description, thank you for flagging that.

I don't know about the name, I'm open to suggestions. I used styles to have it open ended, so that if one day we're about to process things that aren't only classes, the name of the rule doesn't need to change.

For example, we can track a style for data attributes e.g. `[data-foo]{ padding: 0 }`

**Comment by dyc3** [comment]
I'm not sure that makes sense for this rule. The primary purpose of CSS classes is to label which elements get which styles. The existence of attributes, ids, etc. can exist for purposes that are not styling.

**Comment by ematipico** [comment]
Yeah makes sense! Let's update the name of the rule. Should we do the same for the other one?
